src/include/gurt/common.h

```diff
@@ -129,13 +129,13 @@ char *d_realpath(const char *path, char *resolved_path);
  * which are defined in-place or through #define but it will not work with
  * strings which are defined as a char * variable as in that case it'll copy
  * the first 8 bytes.  To avoid this case add a static assert on the size,
- * however this does mean that the macro cannot be used with 8 character
- * strings.
+ * so code that tries to mis-use this macro will fail at compile time.
  */
 
 #define D_STRNDUP_S(ptr, s)						\
 	do {								\
-		_Static_assert(sizeof(s) != sizeof(void *), "do not do this"); \
+		_Static_assert(sizeof(s) != sizeof(void *) ||		\
```

Comment:
should we consider adding
_Static_assert(sizeof(s) == sizeof(void *) && !__builtin....) to D_STRNDUP ? Would it work?

Reply:
Do we need to?  The issue is with the sizeof and D_STRNDUP() doesn't call that.  D_STRNDUP_S() will be slightly more efficient, so we could possibly use _Static_assert() to force use of D_STRNDUP_S() when possible for performance reasons.